db/migrate/20171130120700_create_permissions.rb

```diff
@@ -0,0 +1,21 @@
+class CreatePermissions < ActiveRecord::Migration[5.0]
+  def up
+    create_table :permissions do |t|
+      t.string :regex
+      t.integer :position
+    end
+
+    existing_permissions = Token.all.map { |token| token.permissions.split }.flatten.uniq
```

Comment:
This fails on the call to `token.permissions` due to the `has_many :permissions` association being in place prior to the migration being run. Using `token.attributes['permissions']` works.